ext/GeometryOpsDimensionalDataExt/GeometryOpsDimensionalDataExt.jl

```diff
@@ -14,7 +16,24 @@ function GO.polygonize(A::DD.AbstractDimArray; dims=(DD.X(), DD.Y()), crs=GI.crs
             DD.intervalbounds(DD.set(l, DD.Intervals()))
         end
     end
-    GO.polygonize(bounds_vecs..., DD.AbstractDimArray; crs, kw...)
+
+    # This tree switches on the array type.
+    # If ndims < 2, then we can't polygonize anyway, so we error.
+    # If ndims > 2, then we return a DimArray across the remaining dimensions
+    # not provided in `dims`.  Each slice in `dims` (usually X and Y) 
+    if N < 2
+        error("Array had $N dimensions, but it must have at least two!")
+    elseif N == 2
+        # If ndims == 2, then we polygonize directly, and return the output as requested.
+        Ap = PermutedDimsArray(A, dims)
+        GO.polygonize(bounds_vecs..., Ap; crs, kw...)
+    else
+        map(eachslice(A; dims = DD.otherdims(A, dims))) do a
+            ap = PermutedDimsArray(a, dims)
+            GO.polygonize(bounds_vecs..., a; crs, kw...)
```

Comment:
Can we just pass through the dims kw instead of permuting?

Reply:
The x and y dims need to be in the same order as the lookups though, so I'm not sure how we'd do this...

Reply:
Doesn't it already fix that with `dims` ordering? It probably should...

Maybe it's counter intuitive but reordering dims in the inner loop is essentially free, so usually faster than array wrappers that have a runtime Indirection